Cargo.lock

```diff
@@ -292,6 +324,15 @@ dependencies = [
  "vec_map",
 ]
 
+[[package]]
+name = "concurrent-queue"
+version = "1.2.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "30ed07550be01594c6026cff2a1d7fe9c8f683caa798e12b68694ac9e88286a3"
+dependencies = [
+ "cache-padded",
+]
+
```

Comment:
It's unfortunate this change adds so many dependencies.

Reply:
Yeah. It seems async-channel and async-oneshot were added to be able to remove tokio from the dependencies. Not great...

Reply:
Oh, I didn't notice that. I might be able to trim down the dependencies a bit for default case with tokio. I'll give it a try.

Reply:
Actually, with the `async-channel` and `async-oneshot`, those were added because originally I was going to try and move away from the full `futures` crate toward `futures-lite`. I think I had some issues at one point in the refactoring and never quite got around to that.

But it seems that I can just roll back those particular changes for now and consider whether it's worth investigating later on. I think that `async-*` packages were potentially more performant but it probably isn't even noticeable for an application like LSP because there isn't enough throughput.

Reply:
@ry okay, I was able to remove some of these dependencies. It looks like the remaining ones that were added come from the updated `nom` dependency.